src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import * as api from '../../api'
+
+type Props = {
+  selectedTodo: Todo | null,
+  handleCloseModal: ()=> void,
+}
+
+export const TodoModal: React.FC<Props> = React.memo(({ selectedTodo, handleCloseModal }) => {
+
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+
+  useEffect(() => {
+    if (!selectedTodo) {
+      setUser(null);
+      setLoading(false);
+      return;
+    }
+
+    api.getUser(selectedTodo.userId).then(setUser);
```

Comment:
Consider setting `setLoading(false)` after fetching the user data to ensure the loading state is updated correctly. This will prevent the loader from displaying indefinitely.

Reply:
Fixed